library/include/rocwmma/internal/blend.hpp

```diff
@@ -26,149 +26,92 @@
+                return BlendOp::exec(src0, src1);
             }
-        }
-
-    public:
-        // Sanity checks
-        static_assert((BlendOp::opImpl() == CrossLaneOps::Properties::OP_IMPL_VPERM)
-                          || (BlendOp::opImpl() == CrossLaneOps::Properties::OP_IMPL_VBLEND),
-                      "BlendOp must use vperm or blend backend");
-        static_assert((BlendOp::opId() == CrossLaneOps::Properties::OP_ID_BLEND)
-                          || (BlendOp::opId() == CrossLaneOps::Properties::OP_ID_BLEND_BYTE),
-                      "BlendOp is unsupported");
-
-        template <typename DataT>
-        ROCWMMA_DEVICE static inline DataT exec(DataT const& src0, DataT const& src1)
-        {
-            return blendHelper(src0, src1);
-        }
 
-        template <typename DataT, uint32_t VecSize>
-        ROCWMMA_DEVICE static void exec(VecT<DataT, VecSize>& src0, DataT const& src1)
-        {
-            auto it = makeVectorIterator(src0).begin();
-            static_assert(decltype(it)::range() == VecSize,
-                          "VecSize inconsistent with iterator range");
+            template <typename DataT, uint32_t VecSize>
+            ROCWMMA_DEVICE static void exec(VecT<DataT, VecSize>& src0, DataT const& src1)
+            {
+                auto it = makeVectorIterator(src0).begin();
+                static_assert(decltype(it)::range() == VecSize,
+                              "VecSize inconsistent with iterator range");
 
-            // Loop through entire vector
+                // Loop through entire vector
 #pragma unroll
-            for(uint32_t i = 0; i < VecSize; ++i)
-            {
-                *it = blendHelper(*it, src1);
-                it++;
+                for(uint32_t i = 0; i < VecSize; ++i)
+                {
+                    *it = exec(*it, src1);
+                    it++;
+                }
             }
-        }
 
-        template <typename DataT, uint32_t VecSize>
-        ROCWMMA_DEVICE static void exec(VecT<DataT, VecSize>&       src0,
-                                        VecT<DataT, VecSize> const& src1)
-        {
-            auto       it0 = makeVectorIterator(src0).begin();
-            auto const it1 = makeVectorIterator(src1).begin();
-            static_assert(decltype(it0)::range() == VecSize,
-                          "VecSize inconsistent with iterator range");
+            template <typename DataT, uint32_t VecSize>
+            ROCWMMA_DEVICE static void exec(VecT<DataT, VecSize>&       src0,
+                                            VecT<DataT, VecSize> const& src1)
+            {
+                auto       it0 = makeVectorIterator(src0).begin();
+                auto const it1 = makeVectorIterator(src1).begin();
+                static_assert(decltype(it0)::range() == VecSize,
+                              "VecSize inconsistent with iterator range");
```

Comment:
Also here, it1 range check since this is the entry point to blend for input and output vectors.

Reply:
Here, both vectors are of VecSize, but to be consistent we can extend the check to src1 as well